File: src/main/java/team/zeds/ancientmagic/recipes/AltarRecipe.scala
package team.zeds.ancientmagic.recipes

import net.minecraft.resources.ResourceLocation
import net.minecraft.world.Container
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.crafting.{Ingredient, RecipeSerializer}
import team.zeds.ancientmagic.api.recipe.AMAbstractRecipe
import team.zeds.ancientmagic.init.registries.{AMRecipeTypes, AMRegister}

class AltarRecipe(id: ResourceLocation, ingredient: Ingredient, result: ItemStack, exp: Float, time: Int)
  extends AMAbstractRecipe[Container](AMRecipeTypes.ALTAR_RECIPE, id, ingredient, result, exp, time) {
  override def getSerializer: RecipeSerializer[_] = AMRegister.ALTAR_RECIPE_SERIAL.get()
}

File: srcOld/src/main/java/team/zeds/ancientmagic/recipes/AltarRecipe.kt
package team.zeds.ancientmagic.recipes

import net.minecraft.resources.ResourceLocation
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.crafting.Ingredient
import net.minecraft.world.item.crafting.RecipeSerializer
import team.zeds.ancientmagic.api.recipe.AMAbstractRecipe
import team.zeds.ancientmagic.init.registries.AMRecipeTypes
import team.zeds.ancientmagic.init.registries.AMRegister

class AltarRecipe(id: ResourceLocation, ingr: Ingredient, result: ItemStack, exp: Float, time: Int): AMAbstractRecipe(
    AMRecipeTypes.instance!!.altarRecipe,
    id,
    ingr,
    result,
    exp,
    time
) {
    override fun getSerializer(): RecipeSerializer<*> = AMRegister.ALTAR_RECIPE_SERIAL.get()
}